### PR TITLE
Support backend selector

### PR DESCRIPTION
**Pull request by axinging**

### CODE REVIEW

**Comment by axinging** [comment]
@Linchenn PTAL
@qjia7 @gyagp

File: themes/tfjs/helper/index.js [hunk 1]
```diff
@@ -149,6 +149,20 @@ module.exports = function(hexo) {
       return hexo.locals.toObject().data[`${prefix}/api_manifest`].versions[0];
     },
 
+    isWebGPUSupported: function(version) {
+      // WebGPU is supported since version 4.6.0, the previouse version
+      // is 4.4.0.
+      const nowebgpuVersionArray = [4, 4, 0];
```
Comment (gyagp) [inline themes/tfjs/helper/index.js:155]: It's better to check against [4, 6, 0]

**Comment by axinging** [comment]
@gyagp You comments are resolved, PTAL

File: themes/tfjs/helper/index.js [hunk 2]
```diff
@@ -66,6 +66,28 @@ function isNonBrowserApiPage(path) {
       isTflitePage(path) || isTaskApiPage(path);
 }
 
+// WebGPU is supported since version 4.6.0, for WebGL, first version is 2.0.0.
+const firstWebgpuVersion = '4.6.0';
+const firstWebglVersion = '2.0.0';
+function isVersionSupported(version, oldVersion) {
+  if (version === oldVersion) {
+    return true;
+  }
```
Comment (Linchenn) [inline themes/tfjs/helper/index.js:75]: We could remove this check and the final return returns true.

File: themes/tfjs/layout/partials/apiToc.hbs [hunk 3]
```diff
@@ -17,6 +17,24 @@
...
+      </div>
+      <div class="mdc-menu mdc-select__menu">
+        <ul class="mdc-list mdc-menu__items">
+          {{#if (isWebGPUSupported page.title)}}
+          <li class="mdc-list-item" role="option" tabindex="0" data-version={{this}} backend-name='webgpu' aria-selected>webgpu</li>
+          {{/if}}
+          {{#if (isWebGLSupported page.title)}}
+          <li class="mdc-list-item" role="option" tabindex="0" data-version={{this}} backend-name='webgl'>webgl</li>
+          {{/if}}
+          <li class="mdc-list-item" role="option" tabindex="0" data-version={{this}} backend-name='cpu' {{#unless (isWebGPUSupported page.title)}}aria-selected {{/unless}}>cpu</li>
```
Comment (Linchenn) [inline themes/tfjs/layout/partials/apiToc.hbs:34]: Could you help add WASM backend? Since all the other three backends appears.

**Review by Linchenn — COMMENTED** [review]
Thanks Xu!

**Comment by axinging** [comment]
@Linchenn WASM is also added, PTAL

**Comment by axinging** [comment]
I will close and implement this with a new UI design.

**Pull request closed without merging**